presto-parser/src/main/java/com/facebook/presto/sql/tree/RewritingTraversalVisitor.java

```diff
@@ -0,0 +1,490 @@
+    }
+
+    @Override
+    protected Node visitExcept(Except node, C context)
+    {
+        Relation left = (Relation) process(node.getLeft(), context);
+        Relation right = (Relation) process(node.getRight(), context);
+        return new Except(node.getLocation(), left, right, node.isDistinct());
+    }
+
+    @Override
+    protected Node visitValues(Values node, C context)
+    {
+        List<Expression> rows = processExpressionList(node.getRows(), context);
+        return new Values(node.getLocation(), rows);
+    }
+
+    @Override
+    protected Node visitRow(Row node, C context)
+    {
+        List<Expression> items = processExpressionList(node.getItems(), context);
+        return new Row(node.getLocation(), items);
+    }
+
+    @Override
+    protected Node visitTableSubquery(TableSubquery node, C context)
+    {
+        Query query = (Query) process(node.getQuery(), context);
+        return new TableSubquery(node.getLocation(), query);
+    }
+
+    @Override
+    protected Node visitAliasedRelation(AliasedRelation node, C context)
+    {
+        Relation relation = (Relation) process(node.getRelation(), context);
+        return new AliasedRelation(node.getLocation(), relation, node.getAlias(), node.getColumnNames());
+    }
+
+    @Override
+    protected Node visitSampledRelation(SampledRelation node, C context)
+    {
+        Relation relation = (Relation) process(node.getRelation(), context);
+        Expression sampledPercentage = (Expression) process(node.getSamplePercentage(), context);
+        Optional<List<Expression>> columnsToStratifyOn = Optional.empty();
+        if (node.getColumnsToStratifyOn().isPresent()) {
+            columnsToStratifyOn = Optional.of(processExpressionList(node.getColumnsToStratifyOn().get(), context));
+        }
+        return new SampledRelation(node.getLocation(), relation, node.getType(), sampledPercentage, node.isRescaled(), columnsToStratifyOn);
+    }
+
+    @Override
+    protected Node visitJoin(Join node, C context)
+    {
+        Relation left = (Relation) process(node.getLeft(), context);
+        Relation right = (Relation) process(node.getRight(), context);
+
+        Optional<JoinCriteria> criteria = node.getCriteria();
+        if (criteria.isPresent() && criteria.get() instanceof JoinOn) {
+            Expression expression = (Expression) process(((JoinOn) criteria.get()).getExpression(), context);
+            criteria = Optional.of(new JoinOn(expression));
+        }
+
+        return new Join(node.getLocation(), node.getType(), left, right, criteria);
+    }
+
+    @Override
+    protected Node visitUnnest(Unnest node, C context)
+    {
+        List<Expression> expression = processExpressionList(node.getExpressions(), context);
+
+        return new Unnest(node.getLocation(), expression, node.isWithOrdinality());
+    }
+
+    @Override
+    protected Node visitNode(Node node, C context)
+    {
+        return node;
```

Comment:
This should probably throw, just like `ExpressionTreeRewriter`, otherwise anything not handled could have the wrong behavior:
```java
throw new UnsupportedOperationException("not yet implemented: " + node.getClass().getName());